src/machine/machine_rp2040_gpio.go

```diff
@@ -208,3 +211,196 @@ func (p Pin) Set(value bool) {
 func (p Pin) Get() bool {
 	return p.get()
 }
+
+type PinChange uint8
+
+// Pin change interrupt constants for SetInterrupt.
+const (
+	PinLevelLow PinChange = iota
+	PinLevelHigh
+	// Edge falling
+	PinFalling
+	// Edge rising
+	PinRising
+)
+
+// Callbacks to be called for pins configured with SetInterrupt.
+var (
+	pinCallbacks [32]func(Pin)
+	setInt       [2]bool
+)
+
+// SetInterrupt sets an interrupt to be executed when a particular pin changes
+// state. The pin should already be configured as an input, including a pull up
+// or down if no external pull is provided.
+//
+// This call will replace a previously set callback on this pin. You can pass a
+// nil func to unset the pin change interrupt. If you do so, the change
+// parameter is ignored and can be set to any value (such as 0).
+func (p Pin) SetInterrupt(change PinChange, callback func(Pin)) error {
+	if p > 31 || p < 0 {
+		return ErrInvalidInputPin
+	}
+	events := uint32(1 << change)
+	if callback == nil {
+		// disable current interrupt
+		p.setInterrupt(events, false)
+		pinCallbacks[p] = nil
+		return nil
+	}
+
+	if pinCallbacks[p] != nil {
+		// Callback already configured. Should disable callback by passing a nil callback first.
+		return ErrNoPinChangeChannel
+	}
+	pinCallbacks[p] = callback
+	printIrqReg(defIrq)
+	println("set int")
+	p.setInterrupt(events, true)
+	printIrqReg(defIrq)
+	core := CurrentCore()
+	if setInt[core] {
+		// interrupt has already been set. Exit.
+		println("core set")
+		return nil
+	}
+
+	var intr interrupt.Interrupt
+	switch core {
+	case 0:
+		intr = interrupt.New(rp.IRQ_SIO_IRQ_PROC0, gpioHandleInterrupt)
+		arm.NVIC.ICPR[0].Set(1 << rp.IRQ_SIO_IRQ_PROC0)
+	case 1:
+		intr = interrupt.New(rp.IRQ_SIO_IRQ_PROC1, gpioHandleInterrupt)
+	}
+	println("enable", core)
+	intr.Enable()
+	setInt[core] = true
+	printIrqReg(defIrq)
+	return nil
+}
+
+// gpioHandleInterrupt finds the corresponding pin for the interrupt.
+func gpioHandleInterrupt(intr interrupt.Interrupt) {
+	println("gpioHandle")
+	printIrqReg(defIrq)
+	panic("END")
+	var i uint
+	var flags uint32
+	var irq *irqCtrl
+	switch CurrentCore() {
+	case 0:
+		irq = &ioBank0.proc0IRQctrl
+	case 1:
+		irq = &ioBank0.proc1IRQctrl
+	}
+	// first acquire which interrupt has been triggered.
+	for i = 0; i < 4; i++ {
+		flags = irq.intS[i].Get()
+		if flags != 0 {
+			break
+		}
+	}
+	if flags == 0 {
+		GP2.acknowledgeInterrupt(1 << uint32(PinRising))
```

Comment:
Is this GPIO2? But on Nano RP2040 Connect it's connected to Nina wifi module...

Reply:
Whoops, my bad. Was doing some debugging on my pico and pushed my progress after an exhausting debug session. Will fix